### ISSUE TITLE
Add Spring support for JobExecutor in the dedicated extension

### ISSUE THREAD

**Issue by verdi8** [j-easy/easy-batch#341]
A Spring `FactoryBean `for the `JobExecutor `class could be added to have it injected to run jobs and especially to be shut down when the Spring context is closed.

**Comment by verdi8** [comment]
PR coming soon

**Comment by benas** [comment]
We do not need a factory bean for the `JobExecutor`. One could just declare it as a bean in the application context and use it. See example in the [spring tutotrial](https://github.com/j-easy/easy-batch/blob/master/easybatch-tutorials/src/main/resources/org/easybatch/tutorials/advanced/spring/application-context.xml#L49).

**Comment by verdi8** [comment]
The JobExecutor declared as a bean doesn't follow the Spring bean life cycle for shutdown, does it ? 
It would be similar to what's done in the `ThreadPoolExecutorFactoryBean` or `ScheduledExecutorFactoryBean` (`destroy()` calls `shutdown`). Anyway these later can also be used instead of the JobExecutor in a Spring container.

**Comment by benas** [comment]
> The JobExecutor declared as a bean doesn't follow the Spring bean life cycle for shutdown, does it ?

Good point. In the tutorial, the shutdown was done manually by calling `shutdown`, which is not ideal. We can use `destroy-method="shutdown"` to shut it down automatically when the application context is closed, like:

```xml
<bean id="jobExecutor" class="org.easybatch.core.job.JobExecutor" destroy-method="shutdown"/>
```

I fixed this in the spring tutorial with 507f454f7977717d29ecebaba28fb5c23f3d090a.
